### PR TITLE
#2924 Added a PhaseShifterPSign test case

### PR DESCRIPTION
**Pull request by joyelfeghali**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    simulation tools for power systems.
+-->
+<network xmlns="http://www.itesla_project.eu/schema/iidm/1_0" xmlns:as="http://www.itesla_project.eu/schema/iidm/ext/active-season/1_0" xmlns:bbsp="http://www.itesla_project.eu/schema/iidm/ext/busbarsectionposition/1_0" xmlns:clps="http://www.itesla_project.eu/schema/iidm/ext/current-limits-per-season/1_0" xmlns:cp="http://www.itesla_project.eu/schema/iidm/ext/connectable_position/1_0" xmlns:gapc="http://www.itesla_project.eu/schema/iidm/ext/generator_active_power_control/1_0" xmlns:gec="http://www.itesla_project.eu/schema/iidm/ext/generator_entsoe_category/1_0" xmlns:gs="http://www.itesla_project.eu/schema/iidm/ext/generator_startup/1_0" xmlns:hapc="http://www.itesla_project.eu/schema/iidm/ext/hvdc_angle_droop_active_power_control/1_0" xmlns:hopr="http://www.itesla_project.eu/schema/iidm/ext/hvdc_operator_active_power_range/1_0" xmlns:ld="http://www.itesla_project.eu/schema/iidm/ext/load_detail/1_0" xmlns:sa="http://www.itesla_project.eu/schema/iidm/ext/standby_automaton/1_0" id="ieee14bus" caseDate="2017-06-09T10:14:24.146+02:00" forecastDistance="0" sourceFormat="CIM1">
```
Comment (marcochiaramello) [inline nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm:14]: Please, remove useless iidm extensions

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm (hunk 1 above)
Comment (cossartque) [reply nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm:14]: Done

**Review by cossartque — CHANGES_REQUESTED** [review]

File: nrt/data/DynaFlow/cases.py [hunk 2]
```diff
@@ -184,6 +184,16 @@
 
 test_cases.append((case_name, case_description, job_file, 1, standardReturnCodeType, standardReturnCode))
 
+##############################################################################
```
Comment (cossartque) [inline nrt/data/DynaFlow/cases.py:187]: Have you checked that this nrt works properly ? Thanks

**Review by cossartque — APPROVED** [review]

**Review by cossartque — APPROVED** [review]

File: nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      </generator>
+    </voltageLevel>
+    <voltageLevel id="VoltageLevel2" name="VoltageLevel2" nominalV="400" topologyKind="BUS_BREAKER">
+      <busBreakerTopology>
+        <bus id="Bus2" v="400" angle="0"/>
+      </busBreakerTopology>
+      <load id="Load" name="Load" loadType="UNDEFINED" p0="0" q0="0" bus="Bus2" connectableBus="Bus2" p="0" q="0"/>
+    </voltageLevel>
+    <twoWindingsTransformer id="PhaseShifter" name="PhaseShifter" r="0" x="10" g="0" b="0" ratedU1="400" ratedU2="400" voltageLevelId1="VoltageLevel1" bus1="Bus1" connectableBus1="Bus1" voltageLevelId2="VoltageLevel2" bus2="Bus2" connectableBus2="Bus2" p1="0" q1="0" p2="0" q2="0">
+      <phaseTapChanger lowTapPosition="1" tapPosition="13" regulationMode="CURRENT_LIMITER" regulationValue="770" regulating="true">
```
Comment (cossartque) [inline nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm:31]: What we can do is change CURRENT_LIMITER to ACTIVE_POWER_CONTROL, change the regulationValue (i.e the value for PTarget multiplied by 100 to have it in MW)

File: nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.15"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.175"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.2"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.225"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.25"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.275"/>
+        <step r="0" x="0" g="0" b="0" rho="1.0224948" alpha="-0.3"/>
+      </phaseTapChanger>
+    </twoWindingsTransformer>
+    <twoWindingsTransformer id="PhaseShifterBis" name="PhaseShifterBis" r="0" x="10" g="0" b="0" ratedU1="400" ratedU2="400" voltageLevelId1="VoltageLevel1" bus1="Bus1" connectableBus1="Bus1" voltageLevelId2="VoltageLevel2" bus2="Bus2" connectableBus2="Bus2" p1="0" q1="0" p2="0" q2="0">
```
Comment (cossartque) [inline nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.iidm:60]: same

File: nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.par [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    <par type="DOUBLE" name="load_Beta" value="2.5"/>
+    <reference type="DOUBLE" name="load_P0Pu" origData="IIDM" origName="p_pu"/>
+    <reference type="DOUBLE" name="load_Q0Pu" origData="IIDM" origName="q_pu"/>
+    <reference type="DOUBLE" name="load_U0Pu" origData="IIDM" origName="v_pu"/>
+    <reference type="DOUBLE" name="load_UPhase0" origData="IIDM" origName="angle_pu"/>
+  </set>
+  <set id="PhaseShifter">
+    <par type="DOUBLE" name="phaseShifter_t1st" value="40"/>
+    <par type="DOUBLE" name="phaseShifter_tNext" value="20"/>
+    <par type="DOUBLE" name="phaseShifter_PTarget" value="0.3"/>
```
Comment (cossartque) [inline nrt/data/DynaFlow/PhaseShifterPSign/PhaseShifterPSign.par:47]: We can try to get the PTarget through the IIDM (of course a dev in the DYNDataInterface is needed)

**Review by cossartque — CHANGES_REQUESTED** [review]

**Review by cossartque — COMMENTED** [review]
That is nice work !
Have you checked that all the nrt work ? Thanks

**Comment by cossartque** [comment]
Also, could you rebase ? Thanks

**Review by cossartque — DISMISSED** [review]

**Review by cossartque — APPROVED** [review]

**Pull request merged**